src/components/vm/deleteDialog.jsx

```diff
@@ -137,17 +137,41 @@ export class DeleteDialog extends React.Component {
     delete() {
         const Dialogs = this.context;
         const storage = this.state.disks.filter(d => d.checked);
-        const { vm } = this.props;
+        const { vm, onAddErrorNotification } = this.props;
         const storagePools = getVmStoragePools(vm);
 
         Promise.all(
             (Array.isArray(vm.snapshots) ? vm.snapshots : [])
                     .map(snapshot => snapshotDelete({ connectionName: vm.connectionName, domainPath: vm.id, snapshotName: snapshot.name }))
         )
-                .then(() => domainDelete({ name: vm.name, id: vm.id, connectionName: vm.connectionName, options: { destroy: this.props.vm.state != 'shut off', storage: storage }, storagePools }))
-                .then(() => { Dialogs.close(); cockpit.location.go(["vms"]) })
-                .catch(exc => {
-                    this.dialogErrorSet(cockpit.format(_("VM $0 failed to get deleted"), vm.name), exc.message);
+                .then(() => {
+                    return domainDelete({
+                        name: vm.name,
+                        id: vm.id,
+                        connectionName: vm.connectionName,
+                        live: this.props.vm.state != 'shut off',
+                        storagePools
+                    })
+                            .catch(exc => {
+                                console.warn(cockpit.format(_("VM $0 failed to get deleted: $1"), vm.name, exc.message));
```

Comment:
Please drop the translation of this. The message id is different than the error message below, it's redundant work for translators, and in bug reports we actually want the English messages.

Edit: the following commit fixes that.